ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java

```diff
@@ -134,6 +139,11 @@ public BlockProcessingResult processBlock(
       receipts.add(transactionReceipt);
     }
 
+    final WithdrawalsProcessor withdrawalsProcessor =
+        protocolSchedule.getByBlockHeader(blockHeader).getWithdrawalsProcessor();
+    maybeWithdrawals.ifPresent(
+        withdrawals -> withdrawalsProcessor.processWithdrawals(withdrawals, worldState.updater()));
```

Comment:
Just want to confirm my understanding: if we were pre-shanghai, but had withdrawals, we'd run the NoOpWithdrawalsProcessor and not error?

I think the api validation would prevent this, but wondering if we maybe log a warning in the NoOp just in case?

Reply:
That's right, we would run the NoOpWithdrawalsProcessor.

Changing code around this area to return an Optional<WithdrawalProcessor> so there won't be a NoOpWithdrawalsProcessor at all. So will make that change first.